Bichen07 proposal for Voronoi Diagram

Bichen07 opened the pull request:
<!-- Please fill the following checkboxes -->
---
A good PR should include the following parts:

- [ ] Provide a clear title
- [ ] Provide meaningful descriptions in the PR and the commit logs
- [ ] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [ ] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [ ] Exclude irrelevant files, such as binaries, text files, or dot files
- [ ] Make sure tests/CI pass

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,84 @@
+
+Voronoi Diagrams have applications in various fields, such as spatial analysis, nearest-neighbor searches, and computational physics.
+
+The repository for the project will be housed at: https://github.com/Bichen07/Voronoi-Diagram-for-Spatial-Partitioning
```
The repository name is too long.  Please make it reasonably short.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,84 @@
+Voronoi Diagram for Spatial Partitioning
+
+==========
```
This is not correct rst format.

yungyuc reviewed (changes requested):
When requesting for a next review, please leave a global comment summarizing your change.

- [x] Add an entry in `proposal/README.rst` pointing to your proposal.
- [x] [L9](https://github.com/yungyuc/nsdhw_24au/pull/48/files#diff-ed07af70536dcced965546bcb9a32404aa83edbe8038f1697ed3809a4cfab562R9): Use a short repository and project name.
- [x] Change to use less than 80 characters in a line.  Long lines make it hard to discuss and review.

I will look into the rest after the formatting items are addressed.

Bichen07 commented:
I don't understand the clear meaning of "Add an entry in proposal/README.rst pointing to your proposal.". is that mean add something like "The repository for the project will be housed at: https://github.com/Bichen07/VoronoiHybrid"?
I have done second and third requirements

yungyuc reviewed (changes requested):
> I don't understand the clear meaning of "Add an entry in proposal/README.rst pointing to your proposal.". is that mean add something like "The repository for the project will be housed at: https://github.com/Bichen07/VoronoiHybrid"? I have done second and third requirements

Follow https://github.com/yungyuc/nsdhw_24au/pull/49/files#diff-b021a1aa7c42054b3750ca1748c6eb2aeb73df941fabee6bfc79183e60d9817dR52 to add an entry in `proposal/README.rst` pointing to your proposal.

Bichen07 commented:
I have adjusted my format of term project proposal in my branch Bichen07-project.
I also added URL entry in project/README.rst. I don't sure if it is right to achieve "add an entry in proposal/README.rst pointing to your proposal."?
thank you, teacher.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,126 @@
+* **Github Repo**: https://github.com/Bichen07/VoronoiHybrid
+The Voronoi Diagram is a well know data structure in computetational geometry.
+used for partitioning a plane into regions based on distance to a set of points. 
+Each point (called a seed or site) has a corresponding region consisting of all the points closer to that seed than to any other.
```
The lines are still using too many characters in a line.  Use less than 80 characters.

Please pay attention to the text format.  Most editors can be configured to help you stick with a custom plain text width.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,126 @@
+Voronoi Diagrams are used in diverse fields ranging from geography and robotics to machine learning. 
+Calculating Voronoi diagrams efficiently is crucial in large-scale simulations and real-time applications.
+
+However, creating Voronoi diagrams in Python using libraries like SciPy can become inefficient when applied to large datasets 
```
It will be good if you have references/supports for the slowness of Python implementations for the problem.  It's not a requirement, but will help you understand what you will be doing.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,126 @@
+* **Machine Learning**: For clustering, nearest-neighbor searches, and boundary definitions.
+
+*****************
+System Architecture
```
Adding a prototype of data structures for the Voronoi diagrams will help you understand more about what to do.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,126 @@
+*****************
+The project will use:
+
+* Unit Testing: C++ unit tests will be created using Google Test for validating the core computation of the Voronoi diagrams. 
```
FYI, you may or may not need to use Google-testing for C++.  It depends on how you architecture the code.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,126 @@
+    * Set up project structure and basic Python-C++ integration using Pybind11.
+
+3. Week 3 (10/21): 
+    * Implement core Voronoi diagram algorithm in C++. Test correctness using simple cases.
```
Please add how do you plan to automate the testing with CI.

yungyuc reviewed (changes requested):
> I also added URL entry in project/README.rst. I don't sure if it is right to achieve "add an entry in proposal/README.rst pointing to your proposal."? thank you, teacher.

This is what I want.  Thanks.

---

(Always leave a global comment summarizing your change when requesting reviews.)

* [x] Your PR still contains unrelated changesets.  Please get rid of them by squashing.
* [x] Change to use less than 80 characters in a line.  Long lines make it hard to discuss and review.
* [ ] Add plan for CI.

yungyuc commented:
@Bichen07 Please fix the issues and update as soon as possible.  It is way past the deadline.  I will stop accepting proposal anytime.

Bichen07 commented:
I have fixed the problem. include: 1.Use less than 80 characters2. Adding a prototype of data structures3. add plan to automate the testing with CI
but when I create PR. it show "This branch has conflicts that must be resolved". so I update proposal on Github directly by copy my new version proposal.

yungyuc commented:
Your PR still has unrelated changesets.  Please rebase and get rid of them.  Please also resolve the merge conflict before requesting review.

If you need help, go to the discord channel and/or talk to TA.

Bichen07 commented:
I have commit new version.
if there is any error I still made. I will talk to TA for help.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,180 @@
+    * Set up project structure and basic Python-C++ integration using Pybind11.  
+
+3. Week 3 (10/21): 
+    * Implement core Voronoi diagram algorithm in C++. Test correctness  
```
I have not seen your plan for adding automatic testing and CI.  Please make it clear how and when you will do them.

yungyuc reviewed (changes requested):
> I have commit new version. if there is any error I still made. I will talk to TA for help.

There are still unrelated changesets in your PR history.  Please be pro-active to learn how to use git.  Do speak up in the discord channel.

If you have not sought help from peers, please do so.  @lebr0nli, I wonder if you would have time to help @Bichen07 for his git operations?

Things still need to be fixed:

- [ ] Bad history needs to go away from the PR.
- [ ] [`L146`](https://github.com/yungyuc/nsdhw_24au/pull/48/files#diff-ed07af70536dcced965546bcb9a32404aa83edbe8038f1697ed3809a4cfab562R146): Add plan for automatic testing and CI.

lebr0nli commented:
> @lebr0nli, I wonder if you would have time to help @Bichen07 for his git operations?

Sure, feel free to DM me on Discord.
(I've already joined the Discord server for this class, you should be able to find me by searching my handle.)

But sometimes my responses might be slow when -ENOTIME here. Thanks for understanding! :)

yungyuc commented:
> Sure, feel free to DM me on Discord. (I've already joined the Discord server for this class, you should be able to find me by searching my handle.)

Thanks, @lebr0nli .  @Bichen07, please speak up in the channel so that @lebr0nli and the community can help you.  Slow on learning is the enemy of code development.

> But sometimes my responses might be slow when -ENOTIME here. Thanks for understanding! :)

No problem.

Bichen07 commented:
"Successfully rebased and updated refs/heads/Bichen07-project."
I think I have solve the branch rebase problem( I hope so QQ )

yungyuc commented:
> "Successfully rebased and updated refs/heads/Bichen07-project."
> 
> I think I have solve the branch rebase problem( I hope so QQ )

Please pro-actively make sure you got it right.  Next time when you request my review, it should be perfect.  Please do it as soon as possible.

Bichen07 committed:
Add proposal

Bichen07 commented:
I attended the meet up and solved the problem.

yungyuc commented:
> I attended the meet up and solved the problem.

I guess you are requesting the next review, but it does not read like a request.  Please express in a more explicit way in the future.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,181 @@
+  validating the core computation of the Voronoi diagrams. Python tests will  
+  use pytest to ensure the Python interface behaves correctly and efficiently.  
+
+* Continuous Integration (CI): GitHub Actions will be used to automatically  
```
You should have added an inline annotation indicating the change I requested was added here.  Squashing removed all git history so that I will solely rely on you to tell me what to read.  Otherwise, you force me to read through the whole file and compare from my memory from a week ago.

It is your responsibility as an author to remind the reviews what to review.  Be kind to our reviewers next time.

yungyuc commented on project/Bichen07/README.rst:
```diff
@@ -0,0 +1,181 @@
+.. code:: cpp
+
+   struct VoronoiCell {
+       std::vector<Edge> edges;
```
It is tricky to store edges to define Voronoi cells.  Please pay attention to this as you code.

yungyuc reviewed (approved):
It's OK
